core/test/parser/test_type_price.py

```diff
@@ -0,0 +1,70 @@
+import pytest
+from pydantic import BaseModel, ValidationError
+
+from harambe_core.parser.type_price import ParserTypePrice
+
+
+class _TestModel(BaseModel):
+    value: ParserTypePrice()
+
+
+@pytest.mark.parametrize(
+    "input_value, expected_output",
+    [
+        ("$1,000.00", 1000.00),
+        ("€1.00", 1.0),
+        ("£1,000,000.00", 1000000.00),
+        ("¥1000.00", 1000.00),
+        ("₹1000", 1000.00),
+        (1000, 1000.00),
+        (1000.00, 1000.00),
+        ("-¥1,234.56", -1234.56),
+        ("$0.1", 0.1),
+        ("€.1", 0.1),
+        ("1000", 1000.00),
+        ("$1,234.5678", 1234.5678),
+        ("   $1,234  ", 1234.0),
+        ("   $ 1,234  ", 1234.0),
+        ("1.234 €", 1234.0),
+        ("1.234€", 1234.0),
+        ("€1.234", 1234.0),
+        ("1,234$", 1234.0),
+        ("1,234 $", 1234.0),
+        ("1.234,45", 1234.45),
+        ("1.234.456,00", 1234456.00),
+        ("1.000.000", 1000000.00),
+        ("Starting At 12.99", 12.99),
+        ("From 399.99", 399.99),
+        ("0.0004 $", 0.0004),
+    ],
+)
+def test_flexible_price_success(input_value, expected_output):
+    model = _TestModel(value=input_value)
+    assert model.value == pytest.approx(expected_output)
```

Comment:
why are we doing approx? Whats the limit here

Reply:
i usually do this when comparing floats to avoid any unexpected percision issues